src/main/java/uk/nhs/tis/sync/mapper/DmsMapper.java

```diff
@@ -0,0 +1,32 @@
+package uk.nhs.tis.sync.mapper;
+
+/**
+ * A top level mapper interface to allow generic calling of mappers.
+ *
+ * @param <I> The input type.
+ * @param <O> The output type.
+ */
+public interface DmsMapper<I, O> {
+
+  /**
+   * Convert an entity DTO input to a DMS DTO output.
+   *
+   * @param input The input to convert.
+   * @return The converted DMS DTO.
+   */
+  O toDmsDto(I input);
+
+  /**
+   * Convert from an object to a DMS DTO output.
+   *
+   * @param object The object to convert.
+   * @return The converted object, or null if the object was not of the correct class.
+   */
+  default O objectToDmsDto(Object object) {
+    try {
+      return toDmsDto((I) object);
+    } catch (ClassCastException e) {
+      return null;
+    }
```

Comment:
This seems ok and consistent with the existing expectations.

As getting to this point looks like really unexpected behaviour, logging an error would be a useful distinction.  It seems best pushed out to the caller. What do you think?

Reply:
A log entry makes sense, are you thinking we should let the exception be thrown though? I guess its a scenario that should never happen unless someone messes up the enum params, so maybe letting the ClassCastException be thrown is okay.

Reply:
I'm never keen on declaring unchecked exceptions as thrown. Maybe just take out the catch and add a note in the javadoc?

Reply:
I saw this as one of those "while you're there"*, things to consider.  Leaving as is, if only for now, is also ok for me.

*(looking at other PR comments)